contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_resources_generator.py

```diff
@@ -225,6 +224,28 @@ def gst_v4l2_video_decoder_performance_fakesink(
                 }
             )
 
+    def gst_transform_rotate_and_flip(self, scenario_data: list[dict]) -> None:
+        # Iterate through each encoder plugin configuration
+        for item in scenario_data:
+            encoder_plugin = item.get("encoder_plugin")
+            if not encoder_plugin:
+                continue
+            actions = item.get("actions", [""])
+            resolutions = item.get("resolutions", [""])
+
+            # Use itertools.product to create combinations of all parameters
+            for resolution, action in product(resolutions, actions):
+                config = {
+                    "platform": self._conf_name,
+                    "scenario": self._current_scenario_name,
+                    "encoder_plugin": encoder_plugin,
+                    "width": resolution.get("width"),
+                    "height": resolution.get("height"),
```

Comment:
`.get` doesn't exist on strings. Did you mean to use a list of dicts as the default value for `resolutions`?

Reply:
[ae97ce7](https://github.com/canonical/checkbox/pull/1516/commits/ae97ce7bbfac3176aaeba5efefc4eb78967dc175)

Reply:
Assign the `{}` for resolutions if there's no value be defined in JSON.